samples/system-test/spanner.test.js

```diff
@@ -45,60 +50,68 @@ describe('Spanner', () => {
   before(async () => {
     const instance = spanner.instance(INSTANCE_ID);
     const database = instance.database(DATABASE_ID);
-    try {
-      await instance.delete();
-    } catch (err) {
-      // Ignore error
+
+    if (!INSTANCE_ALREADY_EXISTS) {
+      try {
+        await instance.delete();
+      } catch (err) {
+        // Ignore error
+      }
     }
     try {
       await database.delete();
     } catch (err) {
       // Ignore error
     }
 
-    const [, operation] = await instance.create({
-      config: 'regional-us-central1',
-      nodes: 1,
-      labels: {
-        'gcloud-sample-tests': 'true',
-      },
-    });
-
-    await operation.promise();
-
-    const [instances] = await spanner.getInstances({
-      filter: 'labels.gcloud-sample-tests:true',
-    });
-
-    await Promise.all(
-      instances.map(async instance => {
-        const instanceName = instance.metadata.name;
-        const res = await spanner.auth.request({
-          url: `https://spanner.googleapis.com/v1/${instanceName}/operations`,
-        });
-        const operations = res.data.operations;
-        await Promise.all(
-          operations
-            .filter(operation => {
-              return operation.metadata['@type'].includes('CreateInstance');
-            })
-            .filter(operation => {
-              const yesterday = new Date();
-              yesterday.setHours(-24);
-              const instanceCreated = new Date(operation.metadata.startTime);
-              return instanceCreated < yesterday;
-            })
-            .map(() => instance.delete())
-        );
-      })
-    );
+    if (!INSTANCE_ALREADY_EXISTS) {
+      const [, operation] = await instance.create({
+        config: 'regional-us-central1',
+        nodes: 1,
+        labels: {
+          'gcloud-sample-tests': 'true',
+        },
+      });
+
+      await operation.promise();
+
+      const [instances] = await spanner.getInstances({
+        filter: 'labels.gcloud-sample-tests:true',
+      });
+
+      await Promise.all(
+        instances.map(async instance => {
+          const instanceName = instance.metadata.name;
+          const res = await spanner.auth.request({
+            url: `https://spanner.googleapis.com/v1/${instanceName}/operations`,
+          });
+          const operations = res.data.operations;
+          await Promise.all(
+            operations
+              .filter(operation => {
+                return operation.metadata['@type'].includes('CreateInstance');
```

Comment:
There's a type error that's causing this to fail. The code is not able to access `metadata[@type]` even though the returned operation object has all the relevant fields, this line causes error on the next `filter` call:
```
  1) Spanner
       "before all" hook for "should create an example database":
     TypeError: Cannot read property 'filter' of undefined
```

The metadata object for reference.
```
{ '@type':
   'type.googleapis.com/google.spanner.admin.instance.v1.CreateInstanceMetadata',
  instance:
   { name:
      'projects/my-project/instances/test-instance-1584931910407',
     config:
      'projects/my-project/instanceConfigs/regional-us-central1',
     displayName: 'test-instance-1584931910407',
     nodeCount: 1,
     state: 'READY',
     labels: { 'gcloud-sample-tests': 'true' } },
  startTime: '2020-03-23T02:51:54.268455Z',
  endTime: '2020-03-23T02:51:54.579785Z' }
```

Reply:
Huh! This is also broken on master at least when I try to run the tests locally. Filed #866